ITK_PerformanceTestingDriver.sh

```diff
@@ -0,0 +1,172 @@
+#!/bin/bash
+# \author Hans J. Johnson
+#
+
+#-------------------------------------------------------------------------------------------------------------------
+## https://stackoverflow.com/questions/6481005/how-to-obtain-the-number-of-cpus-cores-in-linux-from-the-command-line
+# macOS:           Use `sysctl -n hw.*cpu_max`, which returns the values of
+#                  interest directly.
+#                  CAVEAT: Using the "_max" key suffixes means that the *maximum*
+#                          available number of CPUs is reported, whereas the
+#                          current power-management mode could make *fewer* CPUs
+#                          available; dropping the "_max" suffix would report the
+#                          number of *currently* available ones; see [1] below.
+#
+# Linux:           Parse output from `lscpu -p`, where each output line represents
+#                  a distinct (logical) CPU.
+#                  Note: Newer versions of `lscpu` support more flexible output
+#                        formats, but we stick with the parseable legacy format
+#                        generated by `-p` to support older distros, too.
+#                        `-p` reports *online* CPUs only - i.e., on hot-pluggable
+#                        systems, currently disabled (offline) CPUs are NOT
+#                        reported.
+
+# Number of LOGICAL CPUs (includes those reported by hyper-threading cores)
+  # Linux: Simply count the number of (non-comment) output lines from `lscpu -p`,
+  # which tells us the number of *logical* CPUs.
+logicalCpuCount=$([[ $(uname) = 'Darwin' ]] &&
+                       sysctl -n hw.logicalcpu_max ||
+                       lscpu -p | egrep -v '^#' | wc -l)
+
+# Number of PHYSICAL CPUs (cores).
+  # Linux: The 2nd column contains the core ID, with each core ID having 1 or
+  #        - in the case of hyperthreading - more logical CPUs.
+  #        Counting the *unique* cores across lines tells us the
+  #        number of *physical* CPUs (cores).
+physicalCpuCount=$([[ $(uname) = 'Darwin' ]] &&
+                       sysctl -n hw.physicalcpu_max ||
+                       lscpu -p | egrep -v '^#' | sort -u -t, -k 2,4 | wc -l)
+#-------------------------------------------------------------------------------------------------------------------
+
+if [[ 0 -eq 1 ]] ;then
+#  Get a long list of hashes to consider
+#  git log v4.12.0..origin/master --reverse --merges --format="%H %ci %aN %s"
+#
+export CC=$(which gcc)
+export CXX=$(which g++)
```

Comment:
@thewtex @dzenanz @blowekamp @jhlegarreta 

This patch should be ready for merging now.  Please review.
 
This script can be run on my computer by copying lines lines 45-54  to my shell,  then it steps through hash-by-hash building ITK, running the test, and storing the results.  I'm creating a performance history from 4.12.0 to present day tonight.

I would appreciate any help I can get for improving this script. 

A big missing piece is how to automatically upload the generate JSON files to a central location for evaluation.
A second missing piece is some analysis code to plot graphs and do statistical analysis.

Reply:
As said, I'd be **pleased to contribute**. I could have a look at the **plot part** (`matplotlib`?) if we're able to specify the variables we'd like to have displayed (at least a shortlist to start with something). We can **open an issue to keep track of this**.

As for **testing this**, do we need to follow the instructions that work on your local workstation?

Reply:
@jhlegarreta Visualization will be fantastic! I think we should use Plot.ly based charts:

  https://plot.ly/pandas/

This will ensure the results can be shared with the community, interacted with, and dynamically regenerated in a web browser so the community can view and track progress. Plot.ly has interop with matplotlib and pandas, so we can start from there.

Here are some good related starting points:

  https://plot.ly/pandas/error-bars/

  https://plot.ly/pandas/axvspan-axvline-axhline/

Reply:
@thewtex Thanks for the link.s I've opened #38